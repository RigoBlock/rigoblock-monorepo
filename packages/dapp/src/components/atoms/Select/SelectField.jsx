import './SelectField.scss'
import 'react-md/lib/Helpers'
import MaterialSelect from 'react-md/lib/SelectFields'
import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'

const SelectField = ({ id, items, placeholder }) => {
  const classProps = classNames('md-cell', 'select-field')
  const dropDownArrow = (
    <span className={'material-icons'}>keyboard_arrow_down</span>
  )
  return (
    <MaterialSelect
      id={id}
      placeholder={placeholder}
      className={classProps}
      menuItems={items}
      position={MaterialSelect.Positions.BELOW}
      dropdownIcon={dropDownArrow}
    />
  )
}

SelectField.propTypes = {
  id: PropTypes.number.isRequired,
  placeholder: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.object).isRequired
}

// ObjectMenuItem.propTypes = {
//   label: PropTypes.string.isRequired,
//   value: PropTypes.object.isRequired
// }

export default SelectField
