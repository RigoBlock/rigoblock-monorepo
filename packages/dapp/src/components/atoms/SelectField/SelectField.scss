@import "../../_settings/variables.scss";

.select-field {
  background: $panels-color;
  border-radius: $wrapper-radius;
  box-shadow: $box-shadow;
}

.select-field-list {
  border-radius: 4px 4px 0 0;
}
