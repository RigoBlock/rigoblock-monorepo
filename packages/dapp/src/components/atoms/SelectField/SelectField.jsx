import './SelectField.scss'
import 'react-md/lib/Helpers'
import MaterialSelect from 'react-md/lib/SelectFields'
import PropTypes from 'prop-types'
import React from 'react'
import classNames from 'classnames'

const SelectField = ({
  id,
  items,
  placeholder,
  onChange,
  value,
  defaultValue
}) => {
  const classProps = classNames('select-field')
  const dropDownArrow = (
    <span className={'material-icons'}>keyboard_arrow_down</span>
  )
  return (
    <MaterialSelect
      id={id.toString()}
      placeholder={placeholder}
      defaultValue={defaultValue}
      value={value}
      onChange={onChange}
      stripActiveItem={false}
      repositionOnScroll={false}
      className={classProps}
      menuItems={items}
      position={MaterialSelect.Positions.BELOW}
      dropdownIcon={dropDownArrow}
    />
  )
}

SelectField.propTypes = {
  id: PropTypes.number.isRequired,
  placeholder: PropTypes.string,
  defaultValue: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
  items: PropTypes.arrayOf(
    PropTypes.oneOfType([
      PropTypes.string,
      PropTypes.number,
      PropTypes.shape({
        label: PropTypes.string.isRequired,
        value: PropTypes.string.isRequired
      })
    ])
  ).isRequired
}

SelectField.defaultProps = {
  placeholder: null,
  defaultValue: '',
  value: '',
  onChange: () => {}
}

export default SelectField
