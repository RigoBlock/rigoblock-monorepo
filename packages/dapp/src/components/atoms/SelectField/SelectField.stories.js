import '../../_settings/_base.scss'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import React from 'react'
import SelectField from './SelectField'

const props = {
  items: ['item1', 'item2', 'item3', 'item4', 'item5', 'item6'],
  placeholder: 'placeholder',
  id: '0',
  meta: {
    touched: null,
    error: null
  },
  input: {
    value: "I don't work",
    onChange: () => null
  }
}

storiesOf('Atoms/SelectField', module)
  .addDecorator(withKnobs)
  .add('default', () => <SelectField {...props} />)
