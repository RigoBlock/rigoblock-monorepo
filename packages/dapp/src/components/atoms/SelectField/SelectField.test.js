import { shallow } from 'enzyme'
import SelectField from './SelectField.jsx'
import toJson from 'enzyme-to-json'

const itemList = ['item1', 2, { label: 'test', value: '1' }]

const props = {
  id: '1',
  items: itemList,
  onChange: jest.fn(),
  defaultValue: 'item1'
}

describe('SelectField component', () => {
  it('renders correctly', () => {
    expect(
      toJson(shallow(createComponentWithProps(SelectField, props)))
    ).toMatchSnapshot()
  })

  it('calls onChange function when value is selected', () => {
    const selectField = shallow(createComponentWithProps(SelectField, props))
    selectField.simulate('change')
    expect(props.onChange).toHaveBeenCalledTimes(1)
  })
})
