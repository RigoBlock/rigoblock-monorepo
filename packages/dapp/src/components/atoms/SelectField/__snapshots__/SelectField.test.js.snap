// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField component renders correctly 1`] = `
<SelectField
  anchor={
    Object {
      "x": "inner left",
      "y": "overlap",
    }
  }
  className="select-field"
  defaultValue=""
  defaultVisible={false}
  dropdownIcon={
    <span
      className="material-icons"
    >
      keyboard_arrow_down
    </span>
  }
  fixedTo={[Window]}
  id="1"
  itemLabel="label"
  itemProps="getProps"
  itemValue="value"
  keyboardMatchingTimeout={1000}
  lineDirection="left"
  listClassName="select-field-list"
  menuItems={
    Array [
      "item1",
      2,
      Object {
        "label": "test",
        "value": 1,
      },
    ]
  }
  onChange={[MockFunction]}
  placeholder="placeholder"
  position="below"
  repositionOnResize={false}
  repositionOnScroll={false}
  transitionName="md-drop"
  transitionTime={300}
/>
`;
